compiler/rustc_parse/src/errors.rs

```diff
@@ -451,6 +451,15 @@ pub(crate) struct MissingExpressionInForLoop {
     pub span: Span,
 }
 
+#[derive(Diagnostic)]
+#[diag(parse_loop_else)]
+#[note]
+pub(crate) struct LoopElseNotSupported {
+    #[primary_span]
+    pub span: Span,
+    pub loop_kind: &'static str,
```

Comment:
I think it's overkill to make this a dedicated enum, since it's not gonna be translated and the variants are few :)